Random Stream closed in multiproject build, Detekt 1.19.0, JDK 1.8

Ran on JDK 1.8, analysing multiple modules in parallel. Probably the standard issue with reusing cached zip streams within the same VM.

```
Caused by: java.io.IOException: Stream closed
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig$getDeprecatedProperties(ValidateConfig.kt:71)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig$testKeys(ValidateConfig.kt:83)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig$testKeys(ValidateConfig.kt:107)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig$testKeys(ValidateConfig.kt:107)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig(ValidateConfig.kt:113)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig(ValidateConfig.kt:34)$
at io.gitlab.arturbosch.detekt.core.config.CompositeConfig.validate(CompositeConfig.kt:31)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig(ValidateConfig.kt:114)$
at io.gitlab.arturbosch.detekt.core.config.ValidateConfigKt.validateConfig(ValidateConfig.kt:34)$
at io.gitlab.arturbosch.detekt.core.config.DefaultPropertiesConfigValidator.validate(DefaultPropertiesConfigValidator.kt:24)$
at io.gitlab.arturbosch.detekt.core.config.ConfigValidatorsKt.checkConfiguration(ConfigValidators.kt:18)$
at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$analyze$1.invoke(Lifecycle.kt:35)$
at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$analyze$1.invoke(Lifecycle.kt:35)$
at io.gitlab.arturbosch.detekt.core.util.PerformanceMonitor.measure(PerformanceMonitor.kt:42)$
at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.measure(Lifecycle.kt:32)$
at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.analyze(Lifecycle.kt:35)$
at io.gitlab.arturbosch.detekt.core.tooling.DefaultLifecycle.analyze(Lifecycle.kt:59)$
at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade$runAnalysis$1.invoke(AnalysisFacade.kt:43)$
at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade$runAnalysis$1.invoke(AnalysisFacade.kt:42)$
at io.gitlab.arturbosch.detekt.core.tooling.ProcessingSpecSettingsBridgeKt.withSettings(ProcessingSpecSettingsBridge.kt:26)$
at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.runAnalysis$detekt_core(AnalysisFacade.kt:42)$
at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.run(AnalysisFacade.kt:25)$
at io.gitlab.arturbosch.detekt.cli.runners.Runner.call(Runner.kt:33)$
at io.gitlab.arturbosch.detekt.cli.runners.Runner.execute(Runner.kt:23)$
at io.gitlab.arturbosch.detekt.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:59)$
at io.gitlab.arturbosch.detekt.Detekt.check(Detekt.kt:252)$
at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)$
```

## Your Environment

* JDK 1.8
* Version of Detekt used: 1.19.0
* Version of Gradle used (if applicable): 7.3.1
* Operating System and version: macOS Monterey

--- a/config/detekt/detekt.yml
+++ b/config/detekt/detekt.yml
@@ -169,8 +169,8 @@ style:
     methods:
       - 'kotlin.io.print'
       - 'kotlin.io.println'
-      - 'java.net.URL.openStream()'
-      - 'java.lang.Class.getResourceAsStream()'
+      - 'java.net.URL.openStream'
+      - 'java.lang.Class.getResourceAsStream'
   ForbiddenVoid:
     active: true
   LibraryCodeMustSpecifyReturnType:

--- a/config/detekt/detekt.yml
+++ b/config/detekt/detekt.yml
@@ -171,6 +171,7 @@ style:
       - 'kotlin.io.println'
       - 'java.net.URL.openStream'
       - 'java.lang.Class.getResourceAsStream'
+      - 'java.lang.ClassLoader.getResourceAsStream'
   ForbiddenVoid:
     active: true
   LibraryCodeMustSpecifyReturnType:

--- a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/config/ValidateConfig.kt
+++ b/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/config/ValidateConfig.kt
@@ -1,5 +1,6 @@
 package io.gitlab.arturbosch.detekt.core.config
 
+import io.github.detekt.utils.openSafeStream
 import io.gitlab.arturbosch.detekt.api.Config
 import io.gitlab.arturbosch.detekt.api.Notification
 import io.gitlab.arturbosch.detekt.api.internal.CommaSeparatedPattern
@@ -67,13 +68,16 @@ internal fun validateConfig(
     val notifications = mutableListOf<Notification>()
 
     fun getDeprecatedProperties(): List<Pair<Regex, String>> {
-        return settings.javaClass.classLoader.getResourceAsStream("deprecation.properties")!!.use { inputStream ->
-            val prop = Properties().apply { load(inputStream) }
+        return settings.javaClass.classLoader
+            .getResource("deprecation.properties")!!
+            .openSafeStream()
+            .use { inputStream ->
+                val prop = Properties().apply { load(inputStream) }
 
-            prop.entries.map { entry ->
-                (entry.key as String).toRegex() to (entry.value as String)
+                prop.entries.map { entry ->
+                    (entry.key as String).toRegex() to (entry.value as String)
+                }
             }
-        }
     }
 
     fun testKeys(current: Map<String, Any>, base: Map<String, Any>, parentPath: String?) {

--- a/detekt-gradle-plugin/build.gradle.kts
+++ b/detekt-gradle-plugin/build.gradle.kts
@@ -45,6 +45,7 @@ configurations.compileOnly { extendsFrom(pluginCompileOnly) }
 dependencies {
     compileOnly(libs.kotlin.gradlePluginApi)
     implementation(libs.sarif4k)
+    implementation(projects.detektUtils)
 
     pluginCompileOnly(libs.android.gradle)
     pluginCompileOnly(libs.kotlin.gradle)

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
@@ -1,5 +1,6 @@
 package io.gitlab.arturbosch.detekt
 
+import io.github.detekt.utils.openSafeStream
 import io.gitlab.arturbosch.detekt.extensions.DetektExtension
 import io.gitlab.arturbosch.detekt.internal.DetektAndroid
 import io.gitlab.arturbosch.detekt.internal.DetektJvm
@@ -120,6 +121,6 @@ const val CONFIGURATION_DETEKT = "detekt"
 const val CONFIGURATION_DETEKT_PLUGINS = "detektPlugins"
 
 internal fun loadDetektVersion(classLoader: ClassLoader): String = Properties().run {
-    load(classLoader.getResourceAsStream("versions.properties")!!)
+    load(classLoader.getResource("versions.properties")!!.openSafeStream())
     getProperty("detektVersion")
 }

--- a/detekt-utils/src/main/kotlin/io/github/detekt/utils/Resources.kt
+++ b/detekt-utils/src/main/kotlin/io/github/detekt/utils/Resources.kt
@@ -19,3 +19,7 @@ fun URL.openSafeStream(): InputStream {
 fun <T> Class<T>.getSafeResourceAsStream(name: String): InputStream? {
     return getResource(name)?.openSafeStream()
 }
+
+fun ClassLoader.getSafeResourceAsStream(name: String): InputStream? {
+    return getResource(name)?.openSafeStream()
+}

